Author: Kaz Kylheku <kkylheku@sierrawireless.com>
Date:   Mon Mar 27 16:49:59 2017 -0700

Patch for use of non-literal format string.

Newer gcc implements this diagnostic for strftime-like
functions?

* glib/gdate.c (g_date_strftime): Wrap function with
pushed/popped #pragma for defeating the warning.

diff --git a/glib/gdate.c b/glib/gdate.c
index 670f7aa..4c84303 100644
--- a/glib/gdate.c
+++ b/glib/gdate.c
@@ -2416,6 +2416,9 @@ win32_strftime_helper (const GDate     *d,
 
 #endif
 
+#pragma GCC diagnostic push /* g_date_strftime uses non-constant format lit */
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+
 /**
  * g_date_strftime:
  * @s: destination buffer
@@ -2549,3 +2552,5 @@ g_date_strftime (gchar       *s,
   return retval;
 #endif
 }
+
+#pragma GCC diagnostic pop
